lib/srv/desktop/windows_server.go

```diff
@@ -405,6 +405,34 @@ func NewWindowsService(cfg WindowsServiceConfig) (*WindowsService, error) {
 		s.cfg.Log.Infoln("desktop discovery via LDAP is disabled, set 'base_dn' to enable")
 	}
 
+	// if LDAP-based discovery is not enabled, but we have configured LDAP
+	// then it's important that we periodically try to use the LDAP connection
+	// to detect connection closure
+	if s.ldapConfigured && len(s.cfg.DiscoveryBaseDN) == 0 {
```

Comment:
This could be `else if s.ldapConfigured` starting [here](https://github.com/gravitational/teleport/blob/b7a5376d429ffe6e7bd203b96172b041fb768534/lib/srv/desktop/windows_server.go#L402) to make it more clear that it's done instead of `startDesktopDiscovery`.

Reply:
I gave this a shot but found that it made the remaining two cases harder to reason about, so I'm going to decline this one unless you feel super strongly.